src/main/java/net/revelc/code/impsort/ImpSort.java

```diff
@@ -126,11 +131,15 @@ public Result parseFile(final Path path, final byte[] buf) throws IOException {
     List<String> fileLines = readAllLines(file);
     ParseResult<CompilationUnit> parseResult =
         new JavaParser(new ParserConfiguration().setLanguageLevel(languageLevel)).parse(file);
-    CompilationUnit unit = parseResult.getResult()
-        .orElseThrow(() -> new ImpSortException(path, Reason.UNABLE_TO_PARSE));
+    CompilationUnit unit = parseResult.getResult().orElseThrow(() -> {
+      parseResult.getProblems().forEach(problem -> logger.error(problem.toString()));
+      return new ImpSortException(path, Reason.UNABLE_TO_PARSE,
+              StringUtils.join(parseResult.getProblems(), System.lineSeparator()));
+    });
     if (!parseResult.isSuccessful()) {
-      parseResult.getProblems().forEach(System.out::println);
-      throw new ImpSortException(path, Reason.PARTIAL_PARSE);
+      parseResult.getProblems().forEach(problem -> logger.error(problem.toString()));
+      throw new ImpSortException(path, Reason.PARTIAL_PARSE,
+              StringUtils.join(parseResult.getProblems(), System.lineSeparator()));
```

Comment:
You ended up doing both suggestions, logging and adding to the exception. I was sort of suggesting one or the other. Given the issues with importing the logger, I don't think we need that, if we have the list of problems in the exception.

Reply:
yes, I add it both in log and exception that make sure user could find error details.

About add problems to exception, it may make error message of exception too long if problems have many elements.
I think print problems in log one by one will be more clear.